### PR TITLE
Inverse Gamma

### PR DESCRIPTION
**Pull request by GeorgeNash**
Inverse Gamma distribution.

Has a very steep gradient curve and long tail, had to bump up the number of samples taken in the tests to get accurate results for mean and standard deviation.

https://en.wikipedia.org/wiki/Inverse-gamma_distribution

### CODE REVIEW

File: keanu-project/src/main/java/io/improbable/keanu/distributions/continuous/InverseGamma.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+        return new Diff(dPda, dPdb, dPdx);
+    }
+
+    public static double logPdf(double a, double b, double x) {
+        return a * Math.log(b) + (-a - 1) * Math.log(x) - Math.log(gamma(a)) - (b / x);
+    }
+
+    public static Diff dlnPdf(double a, double b, double x) {
+        double dPda = -digamma(a) + Math.log(b) - Math.log(x);
```
Comment (gordoncaleb) [inline keanu-project/src/main/java/io/improbable/keanu/distributions/continuous/InverseGamma.java]: It's amazing how much more simple the diff of the log is than just the diff. Good work.

File: keanu-project/src/main/java/io/improbable/keanu/vertices/dbl/probabilistic/InverseGammaVertex.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    private DoubleVertex b;
+    private Random random;
+
+
+    public InverseGammaVertex(DoubleVertex a, DoubleVertex b, Random random) {
+        this.a = a;
+        this.b = b;
+        this.random = random;
+        setParents(a, b);
+        setValue(sample());
```
Comment (gordoncaleb) [inline keanu-project/src/main/java/io/improbable/keanu/vertices/dbl/probabilistic/InverseGammaVertex.java]: After PR #35 we won't set the value from sample anymore.

File: keanu-project/src/test/java/io/improbable/keanu/vertices/dbl/probabilistic/GaussianVertexTest.java [hunk 3]
```diff
@@ -26,14 +26,14 @@
 
     @Before
     public void setup() {
-        random = new Random(1);
+        random = new Random();
```
Comment (gordoncaleb) [inline keanu-project/src/test/java/io/improbable/keanu/vertices/dbl/probabilistic/GaussianVertexTest.java]: Seed the random?

File: keanu-project/src/test/java/io/improbable/keanu/vertices/dbl/probabilistic/InverseGammaVertexTest.java [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+    private static final double DELTA = 0.001;
+
+    @Before
+    public void setup() {
+        random = new Random(3);
+    }
+
+    @Test
+    public void samplingProducesRealisticMeanAndStandardDeviation() {
+        int N = 100000;
```
Comment (gordoncaleb) [inline keanu-project/src/test/java/io/improbable/keanu/vertices/dbl/probabilistic/InverseGammaVertexTest.java]: Can we factor this into ProbabilisticDoubleContract? We seem to be doing this for each vertex.

File: keanu-project/src/test/java/io/improbable/keanu/vertices/dbl/probabilistic/GaussianVertexTest.java (hunk 3 above)
Comment (GeorgeNash) [reply keanu-project/src/test/java/io/improbable/keanu/vertices/dbl/probabilistic/GaussianVertexTest.java]: Ta

File: keanu-project/src/test/java/io/improbable/keanu/vertices/dbl/probabilistic/InverseGammaVertexTest.java (hunk 4 above)
Comment (GeorgeNash) [reply keanu-project/src/test/java/io/improbable/keanu/vertices/dbl/probabilistic/InverseGammaVertexTest.java]: Good idea, will do

**Pull request merged**